File: backend/ApiUrbanPlanning/Infrastructure/Constants/UserRoles.cs
namespace apiUrbanPlanning.Infrastructure.Constants
{
    public static class UserRoles
    {
        public const string Member = "member";
        public const string MunicipalityAdmin = "municipality_admin";
        public const string Operator = "operator";
        public const string SuperAdmin = "super_admin";

        public const string AdminPanel =
            $"{SuperAdmin},{MunicipalityAdmin},{Operator}";
    }
}

File: web/src/lib/userRoles.ts
export const UserRoles = {
  Member: 'member',
  MunicipalityAdmin: 'municipality_admin',
  Operator: 'operator',
  SuperAdmin: 'super_admin',
} as const;

export type UserRole = (typeof UserRoles)[keyof typeof UserRoles];

export function isSuperAdmin(role?: string | null): boolean {
  return role === UserRoles.SuperAdmin;
}

export function isMunicipalityStaff(role?: string | null): boolean {
  return (
    role === UserRoles.MunicipalityAdmin || role === UserRoles.Operator
  );
}

export function canAccessAdminPanel(role?: string | null): boolean {
  return isSuperAdmin(role) || isMunicipalityStaff(role);
}
